pkg/utils/asynciter.go

```diff
@@ -143,6 +142,13 @@ func (ic *AsyncItemsCursor) NextItem() (v3io.Item, error) {
 
 	// Read response from channel
 	resp := <-ic.responseChan
+
+	// Ignore 404s
+	if resp.Error != nil && strings.Contains(resp.Error.Error(), "404") {
+		fmt.Println("Got 404", resp.Error, "input:", resp.Request().Input)
```

Comment:
We really have to fix that logger...

Reply:
Why not to use `ic.logger.Error(...` ?

Reply:
the logger is not initialized there. so you need to either pass a logger every time you create an async iterator the problem is that one of the places doesn't have a logger to pass.
Either way our logging stuff isn't that good and Eran said he want to think what is the best practice and then notify us.. so we might as well wait with that a bit